src/components/waiting-list/Reservations.js

```diff
@@ -1,18 +1,38 @@
-import React from 'react';
+import React from "react";
 
-import ReservationEntry from './ReservationEntry';
-import Data from '../../mocks/ReservationsMock';
+import ReservationEntry from "./ReservationEntry";
+import Data from "../../mocks/ReservationsMock";
 
-import './Reservations.css';
+import "./Reservations.css";
 
 class Reservations extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      data: Data
+    };
+
+    this.getData = this.getData.bind(this);
+    this.removeEntry = this.removeEntry.bind(this);
+  }
 
   getData() {
-    return Data.map((reservationData) => {
-      return <ReservationEntry {...reservationData}/>
+    //TODO: May be posible to iterate only when component loads
+    const { data } = this.state;
+    return data.map((reservationData, index) => {
+      reservationData.index = index;
+      reservationData.removeEntry = this.removeEntry;
+      // eslint-disable-next-line react/jsx-key
+      return <ReservationEntry {...reservationData} />;
     });
   }
 
+  removeEntry(index) {
```

Comment:
Let me know if I got this: a reference to this function is passed to a child of the current component. Through this function the child is allowed to modify its parent state. 

If I'm not wrong, would like to know if this is not an antipattern in react

Reply:
This is the easiest way to update a parent component. There are not many ways to achieve this, but here is where Redux or Context come in handy, because you can update components without worrying who did it.

https://ourcodeworld.com/articles/read/409/how-to-update-parent-state-from-child-component-in-react

Reply:
This will get refactored when adding Redux anyway.

Reply:
Nice! 👍